agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -158,6 +224,24 @@ def on_node_info(topic, info)
       async.start(info)
     end
 
+    def on_ipam_start(topic, data)
+      @ipam_client = IpamClient.new
+      loop do
```

Comment:
`while !ipam_ready` might be a cleaner way to do this.. imho `loop/break` kinda sucks. Then again would it be better to have a separate actor for ipam stuff that could be used for checking readiness (it could also emit events)?